actors/builtin/market/market_actor.go

```diff
@@ -138,14 +139,25 @@ func (a Actor) PublishStorageDeals(rt Runtime, params *PublishStorageDealsParams
 	// This allows us to retain and verify only the client's signature in each deal proposal itself.
 	rt.ValidateImmediateCallerType(builtin.CallerTypesSignable...)
 
+	// resolve worker keys for providers in deals
+	provkeys := resolveControlKeys(rt, params.Deals)
```

Comment:
Are you explicitly trying to support the batch publication of deals for different miners but the same worker? I think that's more complicated than necessary. Without that, the deals must all have the same provider and hence same worker, so we can do a simpler check that the first deal provider has caller == worker, and that all the others match. Only one `RequestMinerControlAddrs` required.

Reply:
Yeah, I think I had it in my head that we allowed clients to submit deals, which would imply you might have a bunch of different miners that the client was submitting a batch of deals for. We don't allow that right now it seems (though the restriction is hidden deep down a few methods away from here). I can definitely simplify this

Reply:
This still needs cleanup (esp for consistency with the comment at 150).